### PR TITLE
Fix: Issue 32:  Verbose logging about useLayoutEffect on server render with web-export with toast component

### PR DESCRIPTION
**Pull request by the-simian**
1. adds a line of code to prevent accessing function on server that causes verbose logging
2. adds a prettierrc.json that matches the existing code style

### CODE REVIEW

**Comment by the-simian** [comment]
The toast component still works normally on the web-export, ios and android builds after this change
image

**Comment by the-simian** [comment]
This resolves https://github.com/mrzachnugent/react-native-reusables/issues/32

**Review by mrzachnugent — CHANGES_REQUESTED** [review]
Thank you for contributing!

Can you add this as a custom hook?

In `~/lib/useIsomorphicLayoutEffect.tsx`, can you add

```jsx
import * as React from 'react';

export const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayoutEffect : useEffect
```

And then use it here?

_FYI: If you were to use `import * as React from 'react'` here, you would get the typescript error: "Cannot assign to 'useLayoutEffect' because it is a read-only property"_

**Comment by the-simian** [comment]
You bet, I think a custom hook is a better approach!

**Comment by the-simian** [comment]
Update: I've added the hook for our own uses in situations where we'd call `useLayoutEffect` effect ourselves. I've left the existing shim in since its in a downstream module and I'm not sure how we'd get our hook in there. I can submit a PR to their repo directly in the future and we can then (probably) remove the shim.

File: lib/useIsomorphicLayoutEffect.tsx [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import * as React from 'react';
+
+export const useIsomorphicLayoutEffect =
+  typeof window !== 'undefined' ? React.useLayoutEffect : React.useEffect;
```
[the remark this answers is not in the document]
Comment (the-simian) [reply lib/useIsomorphicLayoutEffect.tsx]: File is now removed

File: components/ui/toast.tsx [hunk 2]
```diff
@@ -1,30 +1,40 @@
...
 
+/**
+ * This suppresses an annoying, but nonbreaking warning
+ * when accessing `useLayoutEffect`on the server.
+ * See issue: https://github.com/mrzachnugent/react-native-reusables/issues/32
+ */
+if (typeof document === 'undefined') {
+  React.useLayoutEffect = React.useEffect;
+}
+
```
[the remark this answers is not in the document]
Comment (the-simian) [reply components/ui/toast.tsx]: This comment has been updated to reference the issue I opened in `react-native-toast-message`

**Comment by the-simian** [comment]
Here is the referenced issue in the other repo and I've made all the other requested changes:
https://github.com/calintamas/react-native-toast-message/issues/530

**Review by mrzachnugent — APPROVED** [review]
Great! Thank you

**Pull request merged**